frame/xc-asset-config/src/lib.rs

```diff
@@ -0,0 +1,314 @@
+//! # Cross-chain Asset Config Pallet
+//!
+//! ## Overview
+//!
+//! This pallet provides mappings between local asset Id and remove asset location.
+//! E.g. a multilocation like `{parents: 0, interior: X1::(Junction::Parachain(1000))}` could ba mapped to local asset Id `789`.
+//!
+//! Additionally, it stores information whether a foreign asset is supported as a payment currency for execution on local network.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Function
+//!
+//! - `register_asset_location` - used to register mapping between local asset Id and remote asset location
+//! - `set_asset_units_per_second` - registers asset as payment currency and sets the desired payment per second of execution time
+//! - `change_existing_asset_location` - changes the remote location of an existing local asset Id
+//! - `remove_payment_asset` - removes asset from the set of supported payment assets
+//! - `remove_asset` - removes all information related to this asset
+//!
+//! User is encouraged to refer to specific function implementations for more comprehensive documentation.
+//!
+//! ### Other
+//!
+//! `AssetLocationGetter` interface for mapping asset Id to asset location and vice versa
+//! - `get_asset_location`
+//! - `get_asset_id`
+//!
+//! `ExecutionPaymentRate` interface for fetching `units per second` if asset is supported payment asset
+//! - `get_units_per_second`
+//!
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet;
+pub use pallet::*;
+
+#[cfg(any(test, feature = "runtime-benchmarks"))]
+mod benchmarking;
+
+#[cfg(test)]
+pub mod mock;
+#[cfg(test)]
+pub mod tests;
+
+pub mod weights;
+pub use weights::WeightInfo;
+
+#[pallet]
+pub mod pallet {
+
+    use crate::weights::WeightInfo;
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+    use parity_scale_codec::HasCompact;
+    use sp_std::boxed::Box;
+    use xcm::{v1::MultiLocation, VersionedMultiLocation};
+
+    #[pallet::pallet]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    /// Callback definition trait for cross-chain asset registration/deregistration notifications.
+    pub trait XcAssetChangedCallback<T: Config> {
```

Comment:
I believe we could easily drop Callback from name without loss in means.

Reply:
Sure.